plugins/memory/memory_osx

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+# -*- sh -*-
+
+# shellcheck disable=SC2046
+
+: << =cut
+
+=head1 NAME
+
+memory - Plugin to measure memory on osx.
+
+=head1 NOTES
+
+This plugin runs the top command once per interval, to discover memory usage on OSX.
+Contributions are welcome to collect additional memory regions, if possible, such as buffers and caches.
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+if [ "$1" = "autoconf" ]; then
+    if [ "$(uname)" = "Darwin" ]; then
+        echo yes
+        exit 0
+    else
+        echo no
+        exit 0
+    fi
+fi
+
+TOTALMEM=$(sysctl hw.memsize | cut -d" " -f2)
+graphlimit=$(( TOTALMEM/1048576 ))
+
+dehumanise_to_M() {
```

Comment:
Please use base SI units (here: bytes) and enable `graph_scale`.
Users will appreciate this, as soon as 256 TB RAM start to be a commodity :)